### DISCUSSION TITLE
Automatic Conversion of videos across Protocols

### DISCUSSION THREAD

**Discussion by MattWanjia** [bluenviron/mediamtx#3488] [Questions]
### Question

I am feeding into me mediamtx server an rtmp stream and would like to access it in webrtc and rtsp formats. The server is not responding with the videos. Almost as if the endpoints are unreachable.

**Comment by asturl** [comment]
What are your configurations for these paths?
Which video codec is in the rtmp stream?
Are there any log/error messages when you access the paths?

**Reply by MattWanjia** [reply]
Its just a normal rtmp. I disabled authentication. I am using this library called libstreaming to send the rtmp from an android device to my server.
https://github.com/lakeinchina/librestreaming
The paths are not just reachable no logs are offered.

**Reply by moenodedev** [reply]
You're not being helpful. Post your config

**Reply by moenodedev** [reply]
If you stream to rtmp://localhost:1935/rpiCamera the video is available at rtsp://localhost:8554/rpiCamera and rtmp://localhost:1935/rpiCamera and https://localhost:8888/rpiCamera
```
paths:
  cam:
    source: rpiCamera
```

**Comment by MattWanjia** [comment]
Do i need to make any request to the Control API before the rtmp stream is converted to other protocols or is it automamtic?

**Comment by MattWanjia** [comment]
[RTSP] [session 426586e6] created by 192.168.1.220:57868
2024/06/20 10:00:35 INF [RTSP] [conn 192.168.1.220:57868] closed: path of a SETUP request must end with a slash. This typically happens when VLC fails a request, and then switches to an unsupported RTSP dialect
2024/06/20 10:00:35 INF [RTSP] [session 426586e6] destroyed: not in use
2024/06/20 10:00:53 INF [WebRTC] [session dcf1ccca] created by 192.168.1.220:42174
2024/06/20 10:00:53 INF [WebRTC] [session dcf1ccca] closed: no one is publishing to path 'BANKS'
2024/06/20 10:00:56 INF [WebRTC] [session 48f56d11] created by 192.168.1.220:42174
2024/06/20 10:00:56 INF [WebRTC] [session 48f56d11] closed: no one is publishing to path 'BANKS'
2024/06/20 10:00:58 INF [WebRTC] [session 5ccdf42d] created by 192.168.1.220:42174
2024/06/20 10:00:58 INF [WebRTC] [session 5ccdf42d] closed: no one is publishing to path 'BANKS'
2024/06/20 10:01:00 INF [WebRTC] [session 7d736dc5] created by 192.168.1.220:42174
2024/06/20 10:01:00 INF [WebRTC] [session 7d736dc5] closed: no one is publishing to path 'BANKS'
2024/06/20 10:01:02 INF [WebRTC] [session 5c7f5ddc] created by 192.168.1.220:42174
2024/06/20 10:01:02 INF [WebRTC] [session 5c7f5ddc] closed: no one is publishing to path 'BANKS'
2024/06/20 10:01:04 INF [WebRTC] [session b78155ec] created by 192.168.1.220:42174
2024/06/20 10:01:04 INF [WebRTC] [session b78155ec] closed: no one is publishing to path 'BANKS'
2024/06/20 10:01:06 INF [WebRTC] [session 61bd6443] created by 192.168.1.220:42174

Here are the errors i am getting

**Comment by aler9** [comment]
Conversion between protocols is automatic and doesn't require any API call or particular configuration.

Further help cannot be provided since you didn't provide instructions on how to replicate the issue. If you still have problems, create another question and include, as mentioned in the instructions:
* the server version you are using
* precise instructions on how to replicate the problem
* server logs with setting logLevel set to debug
